### PR TITLE
1084 - Adding feature toggle to facilitate PVA2 chatbot migration

### PR DESCRIPTION
**Pull request by alexperson-tw**
## Summary

- We added a new feature toggle that would allow us to switch between connecting to our PVA1 and PVA2 chatbots for testing in dev until we migrate all environments to PVA2 and remove the feature toggle.
- This feature toggle allows us to set our webchat secret associated with one chatbot or another.
- I work for the virtual agent chatbot team and our team builds and maintains the virtual agent chatbot and will go through necessary steps to remove the toggle when it is no longer needed.
- We are trying to migrate from PVA1 to PVA2 as quickly as possible. We will remove the toggle once the solution is tested in dev and deployed to our staging environment.

## Related issue(s)
- This PR is related to the card linked below. This PR is needed so that we can switch between versions of our chatbot in dev with a feature toggle.
- https://app.zenhub.com/workspaces/va-virtual-agent-6011826e423fcf000eb231cb/issues/gh/department-of-veterans-affairs/va-virtual-agent/1084

## Testing done

- Before this change, we would have had to change the webchat secret stored in parameter store to switch between chatbots. With this change, we are able to switch between which secret is used with the feature toggle.
- We have run vets-api and vets-website locally and successfully connected to our chatbots and switched between them using the feature toggle.

## What areas of the site does it impact?
This will impact which version of the chatbot is served on this page of vets-website: https://dev.va.gov/contact-us/virtual-agent/

## Acceptance criteria

- [ ]  Added a new secret and feature toggle to switch between which secret is used

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1204,4 +1204,4 @@ RUBY VERSION
    ruby 3.2.2p53
 
 BUNDLED WITH
-   2.4.10
+   2.4.10
```
Comment (LindseySaari) [inline Gemfile.lock:1207]: Looks like your editor may have removed the newline

**Comment by LindseySaari** [comment]
Can we also add a PR description here aside from the link to the ticket please?

**Comment by alexperson-tw** [comment]
Can you elaborate on what you meant by PR description? I thought that was what the summary was for. I added an additional description in the Related Issues section. Is that what you meant?

File: config/settings.yml [hunk 2]
```diff
@@ -1228,6 +1228,7 @@ covid_vaccine:
 
 virtual_agent:
   webchat_secret: fake_secret
+  webchat_pva2_secret: fake_secret
```
Comment (LindseySaari) [inline config/settings.yml:1231]: Has this been added to the setting.yml definition in the values.yml over in the manifests repo for Vets API?

Comment (alexperson-tw) [reply config/settings.yml:1231]: We added the secret to secrets.yaml and values.yaml in the manifests repo, see [merged PR here](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2217/files)

**Review by lealmendozaalexander — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**